### PR TITLE
Add distribution_id to feature_usage_events_v1

### PR DESCRIPTION
**Pull request by m-d-bowerman**
## Description

This PR adds the client distribution_id. This is necessary to count app push notification events and set to default events to support mobile distribution deals.

Adds engagement with the sponsored tiles feature.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-5013)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/fenix_derived/feature_usage_events_v1/query.sql [hunk 1]
```diff
@@ -938,7 +947,27 @@ SELECT
       WHEN home_page_customize_home_clicked > 0
         THEN client_id
     END
-  ) AS home_page_customize_home_clicked_users
+  ) AS home_page_customize_home_clicked_users,
+  distribution_id,
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/fenix_derived/feature_usage_events_v1/query.sql:951]: this `distribution_id` is a bit hidden right now among all the aggregations. Would it make sense to make it live just under the `is_default_browser`? We could keep all the grouping keys together this way and the query easier to read.

File: sql/moz-fx-data-shared-prod/fenix_derived/feature_usage_events_v1/query.sql [hunk 2]
```diff
@@ -330,7 +331,15 @@ event_ping_clients_feature_usage AS (
...
-    ) AS home_page_customize_home_clicked
+    ) AS home_page_customize_home_clicked,
+    COUNTIF(
+      event_category = 'top_sites'
+      AND event_name = 'contile_click'
+    ) AS top_sites_contile_click,
+    COUNTIF(
+      event_category = 'top_sites'
+      AND event_name = 'contile_impression'
+    ) AS top_sites_contile_impression
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/fenix_derived/feature_usage_events_v1/query.sql:342]:
```suggestion
    ) AS top_sites_contile_impression,
```

File: sql/moz-fx-data-shared-prod/fenix_derived/feature_usage_events_v1/query.sql [hunk 3]
```diff
@@ -30,6 +30,7 @@ client_attribution AS (
     client_id,
     channel,
     adjust_network,
+    distribution_id
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/fenix_derived/feature_usage_events_v1/query.sql:33]:
```suggestion
    distribution_id,
```

**Review by kik-kik — APPROVED** [review]
r+wc

Just some small comments. Please read through before merging.

**Comment by kik-kik** [comment]
One thing we should think about, since we're adding a new grouping value should we recompute partitions that were generated prior to when we merge this change? Or alternatively, should we have a note in the metadata.yaml inside the description indicating that this data was only added after certain date to minimise the risk of confusion when trying to use this data?

**Comment by m-d-bowerman** [comment]
> One thing we should think about, since we're adding a new grouping value should we recompute partitions that were generated prior to when we merge this change? Or alternatively, should we have a note in the metadata.yaml inside the description indicating that this data was only added after certain date to minimise the risk of confusion when trying to use this data?

Yes, I'll do some validation before merging this and update the metadata.

**Comment by kik-kik** [comment]
@m-d-bowerman Are we good to merge this?

**Review by kik-kik — CHANGES_REQUESTED** [review]
Is this the correct events that we want to track?

**Review by kik-kik — APPROVED** [review]

**Pull request merged**